internal/handlers/pg/msg_aggregate.go

```diff
@@ -16,12 +16,126 @@ package pg
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v4"
 
 	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/handlers/common/aggregations"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/handlers/pg/pgdb"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // MsgAggregate implements HandlerInterface.
 func (h *Handler) MsgAggregate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
-	return common.MsgAggregate(ctx, msg)
+	dbPool, err := h.DBPool(ctx)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	// TODO https://github.com/FerretDB/FerretDB/issues/1892
+	common.Ignored(document, h.L, "cursor", "lsid")
+
+	if err = common.Unimplemented(document, "explain", "bypassDocumentValidation", "hint"); err != nil {
+		return nil, err
+	}
+
+	if err = common.Unimplemented(document, "readConcern", "writeConcern"); err != nil {
+		return nil, err
+	}
+
+	common.Ignored(document, h.L, "allowDiskUse", "maxTimeMS", "collation", "comment", "let")
```

Comment:
Just a question: In which cases we ignore and in which we send unimplemented error? Is that because allowDiskUse or comment might be used in some application by default and we want to make transition to FerretDB easy?

Reply:
We ignore everything we can; everything that does not affect the query results. `allowDiskUse` is a pure optimization that we don't support

Reply:
But, uh, `let` should not be there